inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.h

```diff
@@ -55,23 +57,52 @@ class MKLDNNPadNode : public MKLDNNNode {
         size_t workAmount = 0lu;
         size_t lastDstDim = 1lu;
         size_t shift = 0lu;
-        uint8_t sizeData = 1;
-    } params;
-
-    template<typename T>
-    struct PadConstantEmitter {
-        void operator()(MKLDNNPadNode* node) {
-            node->padConstantCommon<T>();
-        }
+        size_t dataSize = 1lu;
+        PadMode padMode;
+    };
+
+    struct PadExecutor {
+        PadExecutor(const PadAttrs& params, const InferenceEngine::SizeVector& srcDims, const InferenceEngine::SizeVector& dstDims);
+        void exec(MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr);
+        ~PadExecutor() = default;
+
+    private:
+        void padConstant(MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr);
+        template<typename T> void padConstantCommon(MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr);
+        void padConstantZero(MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr);
+        void padEdge(MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr);
+        void padReflectOrSymmetric(MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr, const bool isSymmetric = false);
+
+        inline void getDstIdx(const InferenceEngine::SizeVector& indexes, size_t& dstIdx) const;
+
+        struct PadContext {
+            PadExecutor* executor;
+            MKLDNNMemoryPtr srcMemPtr;
+            MKLDNNMemoryPtr dstMemPtr;
+        };
+
+        template<typename T>
+        struct PadConstantEmitter {
+            void operator()(PadContext& ctx) {
+                ctx.executor->padConstantCommon<T>(ctx.srcMemPtr, ctx.dstMemPtr);
+            }
+        };
+
+        PadParams params;
     };
 
-    std::string errorPrefix;
-    static const size_t DATA_ID = 0;
-    static const size_t PADS_BEGIN_ID = 1;
-    static const size_t PADS_END_ID = 2;
-    static const size_t PAD_VALUE_ID = 3;
+    static constexpr size_t DATA_ID = 0lu;
+    static constexpr size_t PADS_BEGIN_ID = 1lu;
+    static constexpr size_t PADS_END_ID = 2lu;
+    static constexpr size_t PAD_VALUE_ID = 3lu;
 
     bool isPadValueSpecified = false;
+
+    MKLDNNMemoryPtr srcMemPtr = nullptr;
+    MKLDNNMemoryPtr dstMemPtr = nullptr;
```

Comment:
Because `getParentEdgeAt()` has checks for correct index of edge. For small networks it has value as you know. But okey, may be this overhead is very small. I removed.